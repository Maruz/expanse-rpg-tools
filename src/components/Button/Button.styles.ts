import { css, SerializedStyles } from '@emotion/react';

export const styles = {
  button: (): SerializedStyles => css({}),
};

/*
 *Unstyled
 *text
 *outlined
 *link?
 *toggle
 *contained
 *fab
 */
