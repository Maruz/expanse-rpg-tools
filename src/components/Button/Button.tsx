/** @jsx jsx */
import { jsx } from '@emotion/react';
import { FC } from 'react';

import { styles } from './Button.styles';

export interface IButton extends React.HTMLAttributes<HTMLElement> {
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  type?: 'button' | 'reset' | 'submit';
}

export const Button: FC<IButton> = ({
  children,
  onClick,
  type = 'button',
  ...props
}) => (
  <button css={styles.button} type={type} onClick={onClick} {...props}>
    {children}
  </button>
);
