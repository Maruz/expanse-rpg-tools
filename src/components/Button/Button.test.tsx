import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import { Button } from './Button';

describe('button', () => {
  it('should match snapshot', () => {
    expect.assertions(1);
    const handleClick = jest.fn();
    const component = renderer.create(
      <Button onClick={handleClick}>Click Me</Button>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render children', () => {
    expect.assertions(1);
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button onClick={handleClick}>Click Me</Button>,
    );
    expect(getByText('Click Me')).toBeInTheDocument();
  });
  it('should call onClick callback', () => {
    expect.assertions(1);
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button onClick={handleClick}>Click Me</Button>,
    );
    fireEvent.click(getByText('Click Me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
