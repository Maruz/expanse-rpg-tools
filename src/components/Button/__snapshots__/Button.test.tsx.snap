// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot 1`] = `
.emotion-0 {
  color: inherit;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
  type="button"
>
  Click Me
</button>
`;
